### GENOMIC ANALYSIS OF THE ADAPTATION OF CLINICAL ENTEROBACTERIA TO THE RESISTANCE PLASMID POXA-48

This repository contains all the code used in the work "Genomic Analysis of the Adaptation of Clinical Enterobacteria to the Resistance Plasmid pOXA-48".
The code developed for each of the analysis is organized in order for the sections in the "Material and methods" part of the main text of the paper. The version of each software is specified in the this section of the article.

Sequences availability: all the data generated and analyzed during this project is available under the Bioproject.

This README.md file is a brief guide that explains the utility of each script developed during the project. The structure of the repository follows the same as the Materials and Methods section in the work for the sake of simplicity.

Additionally to these Python and R scripts, other software have been necessary to thoroughly analyze this data are:
* FastQC (https://github.com/s-andrews/FastQC)
* MultiQC (https://github.com/ewels/MultiQC)
* Trim Galore (https://github.com/FelixKrueger/TrimGalore)
* SPAdes (https://github.com/ablab/spades)
* QUAST (https://github.com/ablab/quast)
* BWA-MEM (https://github.com/lh3/bwa)
* SAMtools (https://github.com/samtools/samtools)
* Breseq (https://github.com/barricklab/breseq)
* Snippy (https://github.com/tseemann/snippy)
* Anvi'o (https://anvio.org/)
* Prokka (https://github.com/tseemann/prokka)
*Roary (https://github.com/sanger-pathogens/Roary)

![fig_workflow_TFM](https://github.com/Plasmidloma/TFM_Adaptation_Enterobacter/blob/5fafdb434a10e1464cbf1342b16513a2d24b3a92/Results/Workflow.png)
Bioinformatic workflow. Summary scheme of the bioinformatic pipeline designed in this analysis. Input/intermediate data is shown in blue, software employed in gray and output results in green. As a brief summary, raw data generated by WGS of samples from previous conjugation and curing studies samples was pre-processed and quality filtered. The clean reads obtained were used in the downstream analyses. To detect genomic changes caused by conjugation or curing, the clean reads were mapped against their corresponding reference
genome and a quality assess was performed. The assemblies obtained were annotated and used for an analysis of plasmid characterization and for variant calling, along with the clean reads. To characterize mutations affecting the same function, pathway or biological process, GO annotations were obtained for functional analysis with the assemblies. Finally, to analyse possible adaptive and parallel evolution, the results of the variant calling were represented by visualizing clean reads in Anviâ€™o, using as a reference the pangenome generated with Roary from the annotations of the assemblie
