# GENOMIC ANALYSIS OF THE ADAPTATION OF CLINICAL ENTEROBACTERIA TO THE RESISTANCE PLASMID POXA-48

This repository contains all the code used in the work "Genomic Analysis of the Adaptation of Clinical Enterobacteria to the Resistance Plasmid pOXA-48".
The code developed for each of the analysis is organized in order for the sections in the "Material and methods" part of the main text of the paper. The version of each software is specified in the this section of the article.

Sequences availability: all the data generated and analyzed during this project is available under the Bioproject.

This README.md file is a brief guide that explains the utility of each script developed during the project. The structure of the repository follows the same as the Materials and Methods section in the work for the sake of simplicity.

Additionally to these Python and R scripts, other software have been necessary to thoroughly analyze this data are:
* FastQC (https://github.com/s-andrews/FastQC)
* MultiQC (https://github.com/ewels/MultiQC)
* Trim Galore (https://github.com/FelixKrueger/TrimGalore)
* SPAdes (https://github.com/ablab/spades)
* QUAST (https://github.com/ablab/quast)
* BWA-MEM (https://github.com/lh3/bwa)
* SAMtools (https://github.com/samtools/samtools)
* Breseq (https://github.com/barricklab/breseq)
* Snippy (https://github.com/tseemann/snippy)
* Anvi'o (https://anvio.org/)
* Prokka (https://github.com/tseemann/prokka)
* Roary (https://github.com/sanger-pathogens/Roary)

![fig_workflow_TFM](https://github.com/Plasmidloma/TFM_Adaptation_Enterobacter/blob/5fafdb434a10e1464cbf1342b16513a2d24b3a92/Results/Workflow.png)
Bioinformatic workflow. Summary scheme of the bioinformatic pipeline designed in this analysis. Input/intermediate data is shown in blue, software employed in gray and output results in green. As a brief summary, raw data generated by WGS of samples from previous conjugation and curing studies samples was pre-processed and quality filtered. The clean reads obtained were used in the downstream analyses. To detect genomic changes caused by conjugation or curing, the clean reads were mapped against their corresponding reference
genome and a quality assess was performed. The assemblies obtained were annotated and used for an analysis of plasmid characterization and for variant calling, along with the clean reads. To characterize mutations affecting the same function, pathway or biological process, GO annotations were obtained for functional analysis with the assemblies. Finally, to analyse possible adaptive and parallel evolution, the results of the variant calling were represented by visualizing clean reads in Anvi’o, using as a reference the pangenome generated with Roary from the annotations of the assemblies.


## From raw reads to variant calling
A quality control (QC) of the raw reads was performed using FastQC v.0.11.9 and the reports were unified using MultiQC v.1.21. Sequence trimming was performed using Trim Galore v.0.6.4 [43]. Parameters were adjusted to discard sequences shorter than 50 bp,
trim low-quality ends and trim Nextera adapters and generate FastQC reports. A final compilation of the reports was performed to assess the quality of the trimming process. The final clean reads were used to perform the subsequent downstream analyses.
To detect the genomic changes occurred during the conjugation or curing process, Variant Calling was performed and the assemblies were needed. The references were sequenced using both Illumina and Nanopore technology and closed by hybrid assembly. Hybrid assemblies require a combination of long and short reads and generate closed assemblies. The hybrid assembly was performed using Unicycler v.0.4.0. For those references and strains that could not be closed, SPAdes genome assembler v.3.15.4 was used. Parameters were adjusted to assemble high-coverage isolate samples and with an automatic coverage cutoff value (–isolate –cov-cutoff auto). The quality of all of the assemblies was checked using QUAST v.5.2.0 and were annotated using the NCBI Prokaryotic Genome Annotation Pipeline (PGAP, v.2023-10-03.build7061) and Prokka v.1.14.5.
https://github.com/Plasmidloma/TFM_Adaptation_Enterobacter/tree/main/Scripts/Ref_assembly_Annot

For the strains which references were closed and for the variant calling against pOXA_48, Breseq v.0.38.1 was used using the default parameters. For the strains examined in Conjugation Subset 1.2, the -p flag was employed to detect low frequency mutations. For those strains which references had not been closed, the software used was Snippy v.4.6.0 with default parameters. At the end, all the results were filtered and merged, the result being a high dimensional data table.
https://github.com/Plasmidloma/TFM_Adaptation_Enterobacter/tree/main/Scripts/VC-events
